src/query/stream/streams.cpp

```diff
@@ -157,37 +158,238 @@ void from_json(const nlohmann::json &data, StreamStatus<TStream> &status) {
+                       auto stream_source_ptr = pulsar_stream.stream_source->Lock();
+                       const auto info = stream_source_ptr->Info(pulsar_stream.transformation_name);
+                       mgp_result_record *record{nullptr};
+                       {
+                         const auto success =
+                             TryOrSetError([&] { return mgp_result_new_record(result, &record); }, result);
+                         if (!success) {
+                           return;
+                         }
+                       }
+
+                       auto service_url_value = GetStringValueOrSetError(info.service_url.c_str(), memory, result);
+                       if (!service_url_value) {
+                         return;
+                       }
+
+                       mgp_list *topic_names{nullptr};
+                       {
+                         const auto success = TryOrSetError(
+                             [&] { return mgp_list_make_empty(info.topics.size(), memory, &topic_names); }, result);
+                         if (!success) {
+                           return;
+                         }
+                       }
+
+                       for (const auto &topic : info.topics) {
+                         auto topic_value = GetStringValueOrSetError(topic.c_str(), memory, result);
+                         if (!topic_value) {
+                           return;
+                         }
+                         topic_names->elems.push_back(std::move(*topic_value));
+                       }
+
+                       procedure::MgpUniquePtr<mgp_value> topics_value{nullptr, mgp_value_destroy};
+                       {
+                         const auto success = TryOrSetError(
+                             [&] { return procedure::CreateMgpObject(topics_value, mgp_value_make_list, topic_names); },
+                             result);
+                         if (!success) {
+                           return;
+                         }
+                       }
+
+                       const auto err1 = mgp_result_record_insert(record, "service_url", service_url_value.get());
+                       const auto err2 = mgp_result_record_insert(record, "topics", topics_value.get());
+
+                       if (err1 != MGP_ERROR_NO_ERROR || err2 != MGP_ERROR_NO_ERROR) {
+                         static_cast<void>(mgp_result_set_error_msg(result, "Unable to set the result!"));
+                         return;
+                       }
+                     },
+                     [proc_name](auto && /*other*/) {
+                       throw QueryRuntimeException("'{}' can be only used for Pulsar stream sources", proc_name);
+                     }},
+                 it->second);
+    };
+
+    mgp_proc proc(proc_name, get_stream_info, utils::NewDeleteResource(), false);
+    MG_ASSERT(mgp_proc_add_arg(&proc, "stream_name", procedure::Call<mgp_type *>(mgp_type_string)) ==
```

Comment:
"stream_name" (as well as topics, bootstrap_servers) is repeated multiple times, can we extract this into constrexpr?

Reply:
result names could make sense :+1: 
for args not sure because they are used in one place